File: Angular/src/app/services/encryption-service.ts
import { Injectable } from '@angular/core';
import CryptoJS from 'crypto-js';

@Injectable({
  providedIn: 'root'
})
export class EncryptionService {
  private readonly SECRET_KEY = CryptoJS.enc.Utf8.parse("12345678901234567890123456789012"); // 32 chars (256-bit)
  private readonly IV = CryptoJS.enc.Utf8.parse("1234567890123456"); // 16 chars (128-bit)

  encrypt(plainText: string): string {
    const encrypted = CryptoJS.AES.encrypt(
      CryptoJS.enc.Utf8.parse(plainText),
      this.SECRET_KEY,
      {
        keySize: 256 / 32,
        iv: this.IV,
        mode: CryptoJS.mode.CBC,
        padding: CryptoJS.pad.Pkcs7
      }
    );
    return encrypted.toString(); // Base64 string
  }

  decrypt(cipherText: string): string {
    const decrypted = CryptoJS.AES.decrypt(
      cipherText,
      this.SECRET_KEY,
      {
        keySize: 256 / 32,
        iv: this.IV,
        mode: CryptoJS.mode.CBC,
        padding: CryptoJS.pad.Pkcs7
      }
    );
    return decrypted.toString(CryptoJS.enc.Utf8);
  }
}

// import AES from 'crypto-js/aes';
// import Utf8 from 'crypto-js/enc-utf8';

// @Injectable({
//   providedIn: 'root'
// })
// export class EncryptionService {
//   private readonly SECRET_KEY = CryptoJS.enc.Utf8.parse("12345678901234567890123456789012"); // 32 chars (256-bit)
//   private readonly IV = CryptoJS.enc.Utf8.parse("1234567890123456"); // 16 chars (128-bit)

//   encrypt(plainText: string): string {
//     const encrypted = AES.encrypt(Utf8.parse(plainText), this.SECRET_KEY, {
//       keySize: 256 / 32,
//       iv: this.IV
//     });

//     return encrypted.toString(); // Base64 string
//   }

//   decrypt(cipherText: string): string {
//     const decrypted = AES.decrypt(cipherText, this.SECRET_KEY, {
//       keySize: 256 / 32,
//       iv: this.IV
//     });

//     return decrypted.toString(Utf8);
//   }
// }

File: API/ARAS.Business/Utility/EncryptionService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Security.Cryptography;
using System.Text;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore.Metadata.Internal;

namespace ARAS.Business.Utility
{
    public static class EncryptionService
    {
        private readonly static string _key  = "12345678901234567890123456789012";
        private readonly static string _iv = "1234567890123456";

        public static string Encrypt(string plainText)
        {
            using var aes = Aes.Create();
            aes.Key = Encoding.UTF8.GetBytes(_key);
            aes.IV = Encoding.UTF8.GetBytes(_iv);
            aes.Mode = CipherMode.CBC;
            aes.Padding = PaddingMode.PKCS7;

            var encryptor = aes.CreateEncryptor(aes.Key, aes.IV);
            var plainBytes = Encoding.UTF8.GetBytes(plainText);

            var encryptedBytes = encryptor.TransformFinalBlock(plainBytes, 0, plainBytes.Length);
            return Convert.ToBase64String(encryptedBytes);
        }

        public static string Decrypt(string cipherText)
        {
            using var aes = Aes.Create();
            aes.Key = Encoding.UTF8.GetBytes(_key);
            aes.IV = Encoding.UTF8.GetBytes(_iv);
            aes.Mode = CipherMode.CBC;
            aes.Padding = PaddingMode.PKCS7;

            var decryptor = aes.CreateDecryptor(aes.Key, aes.IV);
            var cipherBytes = Convert.FromBase64String(cipherText);

            var decryptedBytes = decryptor.TransformFinalBlock(cipherBytes, 0, cipherBytes.Length);
            return Encoding.UTF8.GetString(decryptedBytes);
        }

        public static string URLEncrypt(string plainText)
        {
            using var aes = Aes.Create();
            aes.Key = Encoding.UTF8.GetBytes(_key);
            aes.IV = Encoding.UTF8.GetBytes(_iv);

            var encryptor = aes.CreateEncryptor(aes.Key, aes.IV);
            var plainBytes = Encoding.UTF8.GetBytes(plainText);
            var cipherBytes = encryptor.TransformFinalBlock(plainBytes, 0, plainBytes.Length);

            // Convert to Base64Url (URL-safe)
            string base64 = Convert.ToBase64String(cipherBytes);
            return base64.Replace('+', '-').Replace('/', '_').TrimEnd('=');
        }

        public static string URLDecrypt(string encryptedText)
        {
            // Convert back from Base64Url to Base64
            string base64 = encryptedText.Replace('-', '+').Replace('_', '/');
            switch (base64.Length % 4)
            {
                case 2: base64 += "=="; break;
                case 3: base64 += "="; break;
            }

            var cipherBytes = Convert.FromBase64String(base64);

            using var aes = Aes.Create();
            aes.Key = Encoding.UTF8.GetBytes(_key);
            aes.IV = Encoding.UTF8.GetBytes(_iv);

            var decryptor = aes.CreateDecryptor(aes.Key, aes.IV);
            var plainBytes = decryptor.TransformFinalBlock(cipherBytes, 0, cipherBytes.Length);

            return Encoding.UTF8.GetString(plainBytes);
        }
    }
}
